pmm-app/tests/pages/qanPage.js

```diff
@@ -198,26 +227,70 @@ module.exports = {
       "Details Query Time value Doesn't Match expected " + detailsQueryTimeData.val + ' to contain ' + queryTimeData.val
     );
   },
+  async verifyFiltersSectionIsPresent() {
+    I.waitForElement(this.fields.filterCheckboxSelector, 30);
+    I.seeElement(this.fields.filterCheckboxSelector);
+  },
+  async verifyColumnIsNotAvailable(columnName) {
+    I.click(this.fields.addColumnSelector);
+    I.dontSeeElement(`//ul/li[@label='${columnName}']`);
+  },
+  async verifySelectedPageIs (pageNumber) {
+    I.seeElement(`.ant-pagination-item-active[title="${pageNumber}"]`);
+  },
+  async verifyMetricsMatch(rowNumber, dataColumnNumber) {
+    const cellSelector = this.overviewMetricCellValueLocator(rowNumber, dataColumnNumber);
 
+    this.showTooltip(rowNumber, dataColumnNumber);
+    let qpsMetricValue = await I.grabTextFrom(cellSelector);
+    let qpsTooltipValue = await I.grabTextFrom(this.qpsTooltipValueSelector);
+    assert.equal(qpsMetricValue.replace(/[^0-9.]/g,""), qpsTooltipValue.replace(/[^0-9.]/g,""));
+  },
   async clearFilters() {
     let numOfElements = await I.grabNumberOfVisibleElements(this.fields.filterSelection);
     for (let i = 1; i <= numOfElements; i++) {
       I.click(this.fields.filterSelection + '[' + i + ']');
       I.waitForInvisible(this.fields.detailsTable, 30);
     }
   },
-  changeGroupBy(groupBy) {
+  changeGroupBy(groupBy= 'Client Host') {
     I.waitForElement(this.fields.groupBySelector, 30);
-    I.click(this.fields.groupBySelector);
+    I.forceClick(this.fields.groupBySelector);
     I.click(`//ul/li[@label='${groupBy}']`);
   },
   groupByIs(groupBy) {
-    I.waitForElement(`//span[text()='Group by ${groupBy}']`, 30);
-    I.seeElement(`//span[text()='Group by ${groupBy}']`);
+    I.waitForElement(`[data-qa="group-by"] [title="${groupBy}"]`, 30);
+    I.seeElement(`[data-qa="group-by"] [title="${groupBy}"]`);
+  },
+  changeSorting(columnNumber, sortDirection = 'down') {
+    const sortingBlockSelector = this.overviewMetricSortingLocator(columnNumber);
+    I.waitForElement(sortingBlockSelector, 30);
+    I.scrollTo(sortingBlockSelector);
+    I.forceClick(sortingBlockSelector);
+    if (sortDirection === 'up') {
+      I.forceClick(sortingBlockSelector);
+    }
+    I.wait(5);
```

Comment:
can we omit hardcoded wait here?

Reply:
Wait is used because after click element is unavailable and table is reloading. On average, it takes about 4 seconds. I was thinking about using of waitForQANPageLoaded it doesn't work with current page structure. Also now I realized that one more wait is need here to be able to click sorting twice, therefore I add it.